### ISSUE TITLE
IDNet

### ISSUE THREAD

**Issue by lidong04** [ShuangLI59/person_search#7]
Hi @Cysu and @ShuangLI59 ,

I find your work "End-to-End Deep Learning for Person Search" is very interesting and trying to reproduce the result with both joint and Deep+IDNet settings. However, I have some questions about IDNet:
1) For identity classification, does it use cropped pedestrian image (resized to square?) as input image, or uses the whole image (with ground truth bbox for cropped region) as input?
2) In IDNet's output, the number of class is 5532 ids or 5533 (ids + background)?
3) I think training IDNet (only has several samples per each person id) is hard, does it also need the pretrain method and RSS layer mentioned in paper?

Many thanks for your elaboration, I really appreciate it.

Regards,
Dong Li

**Comment by Cysu** [comment]
1. We use the cropped and resized (256x256) pedestrian images as inputs. [related code](https://github.com/ShuangLI59/person_search/blob/master/scripts/make_db.sh#L26-L29)
2. It is [5533](https://github.com/ShuangLI59/person_search/blob/master/models/psdb/VGG16/pretrain.prototxt#L542)
3. Yes, but the ID pretraining itself is much easier than the joint training process. As you can have ~256 different class samples in each minibatch, while in the joint training process, this number is typically less than 10.

**Comment by lidong04** [comment]
@Cysu Thanks a lot for your prompt reply! 
I'll try it out in this way - please correct me if something wrong: 
step1. pretrain: finetune IDNet (with plain softmax layer) from ImageNet pretrained VGG16 model, batchsize ~256;
step2. train: finetune IDNet (with RSS layer) from pretrained model in step1, batchsize ~256;

**Comment by Cysu** [comment]
Yes. You're right. Maybe the batch size could be tweaked. We used 128 in the original experiments.

**Comment by huakaijinse** [comment]
Thank you for sharing. I have some questions about person_search. Does the class of background is a must? Can this adding the background class improve the re-id accuracy?

**Comment by Cysu** [comment]
We hypothesize that having the background class can improve the performance, as the proposals will inevitably contain false positives. Will conduct some experiments to verify this. Thank you very much for the question.
